### DISCUSSION TITLE
Cannot find module 'wagmi' or its corresponding type declarations.ts(2307)

### DISCUSSION THREAD

**Discussion by uzair-cbt** [rainbow-me/rainbowkit#1944] [Q&A] [answered]
I am working NextJs 14 with Typescript "^5". This is my config file

"use client";

import { WagmiProvider } from "wagmi";
import "@rainbow-me/rainbowkit/styles.css";
import { QueryClientProvider, QueryClient } from "@tanstack/react-query";
import { mainnet, polygon, optimism, arbitrum, base } from "wagmi/chains";
import { getDefaultConfig, RainbowKitProvider } from "@rainbow-me/rainbowkit";

const config = getDefaultConfig({
  appName: "V Limited",
  projectId: "**************",
  chains: [mainnet, polygon, optimism, arbitrum, base],
  ssr: true,
});

const Configurations = ({ children }) => {
  const queryClient = new QueryClient();
  return (
    <WagmiProvider config={config}>
      <QueryClientProvider client={queryClient}>
        <RainbowKitProvider>{children}</RainbowKitProvider>
      </QueryClientProvider>
    </WagmiProvider>
  );
};

export default Configurations;

it's not working with typescript.

**Comment by c0rdeiro** [comment]
in that code you're only returning children, so you're missing all the providers.
it should look like this (as per [docs](https://www.rainbowkit.com/docs/installation)

```ts
return <WagmiProvider config={config}>
      <QueryClientProvider client={queryClient}>
        <RainbowKitProvider>
          {children}
        </RainbowKitProvider>
      </QueryClientProvider>
    </WagmiProvider>
```

**Reply by uzair-cbt** [reply, marked as answer]
Issue was fixed. It was typescript version error with wagmi.

 "viem": "^2.8.12",
 "wagmi": "^2.5.11",
 "typescript": "5.4.2"
  
  with these versions it's working fine.

**Comment by 0xmkh** [comment]
@uzair-cbt I can't reproduce this on my end. Can you please share a small reproducible example ?

**Reply by uzair-cbt** [reply]
Issue was fixed. It was typescript version error with wagmi.

 "viem": "^2.8.12",
 "wagmi": "^2.5.11",
 "typescript": "5.4.2"
  
  with these versions it's working fine.
